drivers/clock_control/clock_stm32g0.c

```diff
@@ -27,31 +26,56 @@
 #define pllr(v) z_pllr(v)
 
 /**
- * @brief Set up pll configuration
+ * @brief Return PLL source
  */
-int config_pll_sysclock(void)
+__unused
+static uint32_t get_pll_source(void)
 {
-	uint32_t pll_source, pll_m, pll_n, pll_r;
-
-	pll_n = STM32_PLL_N_MULTIPLIER;
-	pll_m = pll_div(STM32_PLL_M_DIVISOR);
-	pll_r = pllr(STM32_PLL_R_DIVISOR);
-
 	/* Configure PLL source */
 	if (IS_ENABLED(STM32_PLL_SRC_HSI)) {
-		pll_source = LL_RCC_PLLSOURCE_HSI;
+		return LL_RCC_PLLSOURCE_HSI;
 	} else if (IS_ENABLED(STM32_PLL_SRC_HSE)) {
-		pll_source = LL_RCC_PLLSOURCE_HSE;
-	} else {
-		return -ENOTSUP;
+		return LL_RCC_PLLSOURCE_HSE;
 	}
 
-	LL_RCC_PLL_ConfigDomain_SYS(pll_source, pll_m, pll_n, pll_r);
+	__ASSERT(0, "invalid source");
+	return 0;
+}
+
+/**
+ * @brief Set up pll configuration
+ */
+__unused
+void config_pll_sysclock(void)
+{
+	LL_RCC_PLL_ConfigDomain_SYS(get_pll_source(),
+				    pll_div(STM32_PLL_M_DIVISOR),
+				    STM32_PLL_N_MULTIPLIER,
+				    pllr(STM32_PLL_R_DIVISOR));
 
 	LL_RCC_PLL_EnableDomain_SYS();
+}
 
-	return 0;
+
+/**
+ * @brief Return pllout frequency
+ */
+__unused
+uint32_t get_pllout_frequency(void)
+{
+	uint32_t pll_source;
+
+	pll_source = get_pll_source();
+	if (pll_source < 0) {
```

Comment:
`if (pll_source < 0)` is always `false`.
Better to use `if (pll_source == 0)`.

Also, L4 and L5 define `LL_RCC_PLLSOURCE_NONE` that is `0`.

Reply:
Fixed